### PR TITLE
WIP: Remove Type.Class and Type.Anon

### PR DESCRIPTION
**Pull request by JAForbes**
- Add argument for documentation url.

#14

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -31,21 +31,21 @@ function Setup(T, { checkTypes , _env=[] }){
...
+            : constr === Number
+                ? T.Number
+            : constr === Boolean
+                ? T.Boolean
+            : constr === Object
+                ? T.Object
+            : constr === Array
+                ? T.Array(T.Any)
+            : constr === Function
+                ? T.AnyFunction
```
Comment (davidchambers) [inline lib/index.js]: Do you plan to remove these mappings one day?

Comment (JAForbes) [inline lib/index.js]: Yes.  Probably in a separate PR though.

Comment (JAForbes) [inline lib/index.js]: I like the ease of use of paldepind's library,  I think I'd eventually like the users to be able to get value out this library without necessarily being aware or understanding what sanctuary-def is, while at the same time encouraging them to explore that ecosystem.

So if I were to remove these mappings, I'd probably keep inline predicates as a compromise, and potentially export $.Number, etc.  Not sure what the best solution is for that problem.  If you have any good ideas, please let me know 😄

Comment (davidchambers) [inline lib/index.js]: If you want to accept sanctuary-def types, type representatives (such as `Array` and `Function`), and predicates, I suggest providing three separate functions for constructing a sum type. As a user I find it confusing that `$.String` and `String` can be used more or less interchangeably (as can `Array` and `Array.isArray`). That said, it is (unfortunately in my view) idiomatic for JavaScript functions to accept values of all sorts of different types. ;)

Comment (JAForbes) [inline lib/index.js]: Yeah I get your argument.  I think though you can't clearly delineate "this type takes predicates, this type accepts types", often a user will want to do both.  Maybe we could provide a utility for easily creating anonymous predicate types, I'd be happy if those predicate's types' names were just the function body.  e.g. `$.NullaryType( 'x => x % 2 == 0', '', x => x % 2 == 0)` could just become `predicate( x => x % 2 )`.

Comment (davidchambers) [inline lib/index.js]: I like your `predicate` suggestion. :)

**Comment by JAForbes** [comment]
Closing as this all happens in the next major release in one PR (that I'm about to open)

**Pull request closed without merging**